polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -316,24 +304,23 @@ function IndexTableBase({
         }
 
         // update left offset for first column
-        if (selectable && tableHeadings.current.length > 1)
+        if (
+          selectable &&
+          tableHeadings.current.length > 1 &&
+          stickyTableHeadings.current.length > 1
```

Comment:
There is! Sticky table header isn't rendered when `loading` is true (but normal header is... Not sure why there's a difference there, but there is).

Reply:
It would make sense that we want to be able to change the first column offset for the regular header regardless of if the sticky header is present or not, so I think this if needs to be split into 2.